### PR TITLE
Regularize edge normals

### PR DESCRIPTION
**Pull request by jrgnicho**
This PR attempts to set the normal vectors of edge path points if the surface happens to be flat enough.
@Jmeyer1292 please review

### CODE REVIEW

File: godel_surface_detection/include/segmentation/surface_segmentation.h [hunk 1]
```diff
@@ -119,6 +121,29 @@ class SurfaceSegmentation
...
+   * @param boundary_pts                The boundary points
+   * @param poses                       Output argument that contains the poses with the computed normals
+   * @param eps                         Radius used to search for points in the surface near the boundary
+   * @param plane_max_dist              Max distance at which points are considered to be in the plane.
+   * @param plane_inlier_threshold      Minimum ratio of points needed to be confident that the surface is flat.
+   * @return
+   */
+  bool regularizeNormals(const std::vector<pcl::PointNormal>& boundary_pts,
+                         std::vector<Eigen::Matrix4d, Eigen::aligned_allocator<Eigen::Matrix4d>>& poses,
+                         double eps = 0.01,
```
Comment (AustinDeric) [inline godel_surface_detection/include/segmentation/surface_segmentation.h:144]: make const and pass to function.

File: godel_surface_detection/src/segmentation/surface_segmentation.cpp [hunk 2]
```diff
@@ -29,13 +38,31 @@ SurfaceSegmentation::SurfaceSegmentation(pcl::PointCloud<pcl::PointXYZRGB>::Ptr
...
   pcl::copyPointCloud(*icloud, *input_cloud_);
+
+  // downsampling cloud
+  input_cloud_downsampled_.reset(new pcl::PointCloud<pcl::PointXYZ>());
+  pcl::copyPointCloud(*input_cloud_,*input_cloud_downsampled_);
+  pcl::VoxelGrid<pcl::PointXYZ> vg;
+  vg.setInputCloud(input_cloud_downsampled_);
+  vg.setLeafSize (DOWNSAMPLING_LEAF,DOWNSAMPLING_LEAF,DOWNSAMPLING_LEAF);
+  int start_size = input_cloud_downsampled_->size();
+  int end_size = 0;
```
Comment (AustinDeric) [inline godel_surface_detection/src/segmentation/surface_segmentation.cpp]: for dev only, please remove.2

File: godel_surface_detection/src/segmentation/surface_segmentation.cpp [hunk 3]
```diff
@@ -310,11 +337,197 @@ void SurfaceSegmentation::smoothPointNormal(std::vector<pcl::PointNormal> &pts_i
...
 
 
+bool SurfaceSegmentation::regularizeNormals(const std::vector<pcl::PointNormal>& boundary_pts,
+                                            std::vector<Eigen::Matrix4d, Eigen::aligned_allocator<Eigen::Matrix4d>>& poses,
+                                            double eps,
+                                            double plane_max_dist,
+                                            double plane_inlier_threshold)
+{
+  // aliases
+  using PCLPoint = pcl::PointXYZRGB;
```
Comment (AustinDeric) [inline godel_surface_detection/src/segmentation/surface_segmentation.cpp:347]: not a fan of this but its ok.

**Comment by jrgnicho** [comment]
@AustinDeric did you submit the review?

**Review by AustinDeric — CHANGES_REQUESTED** [review]

**Comment by jrgnicho** [comment]
@AustinDeric  I made the requested changes.

**Review by AustinDeric — APPROVED** [review]

**Pull request merged**